lib/algos/list/delimiterMaster.js

```diff
@@ -130,6 +131,9 @@ class DelimiterMaster extends Delimiter {
                 return FILTER_ACCEPT;
             }
             this.prvKey = key;
+            if (!this.allowDelimiterRangeSkip(key)) {
+                this[this.nextContinueMarker] = key;
```

Comment:
The problem I see with this piece of code is that it will prevent skipping over subprefixes when some/all keys are delete markers in those subprefixes, leading to a potentially very long listing or timing out.

Note that today we have a similar issue in v0 but only when delete marker keys are not part of a subprefix, we still have the ability to skip a whole subprefix.

Here's a test that catches this issue:

```
            it.only('foobar', () => {
                for (const prefix of ['prefix/']) {
                    const delimiter = new DelimiterMaster({
                        prefix,
                        delimiter: '/',
                    }, fakeLogger, vFormat);
                    for (const testEntry of [
                        {
                            key: prefix,
                            expectedRes: FILTER_ACCEPT,
                            expectedSkipping: `${prefix}${VID_SEP}`,
                        },
                        {
                            key: `${prefix}${VID_SEP}v1`,
                            //isDeleteMarker: true,
                            expectedRes: FILTER_SKIP, // versions get skipped after master
                            expectedSkipping: `${prefix}${VID_SEP}`,
                        },
                        {
                            key: `${prefix}foo/`,
                            isDeleteMarker: true,
                            expectedRes: FILTER_SKIP,
                            expectedSkipping: `${prefix}foo/`,
                        },
                        {
                            key: `${prefix}foo/1`,
                            isDeleteMarker: true,
                            expectedRes: FILTER_SKIP,
                            expectedSkipping: `${prefix}foo/`,
                        },
                        {
                            key: `${prefix}foo/2`,
                            isDeleteMarker: true,
                            expectedRes: FILTER_SKIP,
                            expectedSkipping: `${prefix}foo/`,
                        },
                    ]) {
                        const entry = {
                            key: testEntry.key,
                        };
                        if (testEntry.isDeleteMarker) {
                            entry.value = '{"isDeleteMarker":true}';
                        } else {
                            entry.value = '{}';
                        }
                        const res = delimiter.filter(entry);
                        const skipping = delimiter.skipping();
                        assert.strictEqual(res, testEntry.expectedRes);
                        assert.strictEqual(skipping, testEntry.expectedSkipping);
                    }
                }
            });
```